packages/lib/services/plugins/api/JoplinClipboard.ts

```diff
@@ -59,4 +61,36 @@ export default class JoplinClipboard {
 		return this.electronClipboard_.availableFormats();
 	}
 
+	/**
+	 * Writes multiple formats to the clipboard simultaneously.
+	 * This allows setting both text/plain and text/html at the same time.
+	 *
+	 * <span class="platform-desktop">desktop</span>
+	 *
+	 * @example
+	 * ```typescript
+	 * await joplin.clipboard.writeMultiple({
+	 *   text: 'Plain text version',
+	 *   html: '<strong>HTML version</strong>'
+	 * });
+	 * ```
+	 */
+	public async writeMultiple(content: ClipboardContent): Promise<void> {
+		const clipboardData: Record<string, unknown> = {};
+
+		if (content.text !== undefined) {
+			clipboardData.text = content.text;
+		}
+
+		if (content.html !== undefined) {
+			clipboardData.html = content.html;
+		}
+
+		if (content.image !== undefined) {
+			clipboardData.image = this.electronNativeImage_.createFromDataURL(content.image);
+		}
+
+		this.electronClipboard_.write(clipboardData);
```

Comment:
If nothing was written, maybe skip this call? Unless we want to support clearing the clipboard, but I'm not sure we want this as that might be unexpected (we just merge two PR to avoid clearing the clipboard when copying nothing from the editor)